### PR TITLE
Fixes for settings resource

### PR DESCRIPTION
**Pull request by kirrg001**
refs #10582 

- [x] more manual testing
- [x] check settings cache - does it break anywhere?
- [x] ensure we won't break v0.1

### CODE REVIEW

File: core/test/regression/api/v2/admin/db_spec.js [hunk 1]
```diff
@@ -65,6 +67,47 @@ describe('DB API', () => {
...
+                    .expect(200);
+            })
+            .then((res) => {
+                const jsonResponse = res.body;
+                should.exist(jsonResponse.db);
+
+                const exportFolder = path.join(os.tmpdir(), uuid.v1());
+                const exportPath = path.join(exportFolder, 'export.json');
+                fs.ensureDirSync(exportFolder);
+                fs.writeJSONSync(exportPath, jsonResponse);
```
Comment (gargol) [inline core/test/regression/api/v2/admin/db_spec.js]: Added a small commit that also does cleanup for this folder :wink:

File: core/server/models/settings.js [hunk 2]
```diff
@@ -125,10 +124,6 @@ Settings = ghostBookshelf.Model.extend({
             attrs.value = JSON.parse(attrs.value);
         }
 
-        if (attrs.value === 'null') {
-            attrs.value = null;
```
Comment (kirrg001) [inline core/server/models/settings.js]: @gargol Could you please confirm & double check in the morning? 

1. I am not aware of a report where people have "null" in their database, but it means NULL
2. I am not aware of a use case.

Maybe I forgot a use case/report.

**Comment by kirrg001** [comment]
Waiting with merging because Travis is broken currently (see master). We are working on some improvements for Travis.

File: core/server/models/settings.js (hunk 2 above)
Comment (gargol) [reply core/server/models/settings.js]: @kirrg001 Found one case where we end up with `"null"` value in the db. To reproduce it you need to roll back to `2.16.3` and do an import of previously public site (`is_private` setting flag has to be falsy). After import we end up having:
```
mysql> select `key`, `value` from `settings` where `value` = 'null';
+----------+-------+
| key      | value |
+----------+-------+
| password | null  |
+----------+-------+
```

Comment (gargol) [reply core/server/models/settings.js]: We should probably include a fix for this in our migrations as well?

Comment (kirrg001) [reply core/server/models/settings.js]: Thanks for testing btw :)

Comment (gargol) [reply core/server/models/settings.js]: Pretty sure it was `null` as returned from `grep -roP "password.{0,15}" ./import.json`:
```
password","value":null,
```

**Pull request merged**